packages/core/strapi/lib/commands/__tests__/export.test.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const utils = require('../transfer/utils');
+
+const mockDataTransfer = {
+  createLocalFileDestinationProvider: jest.fn(),
+  createLocalStrapiSourceProvider: jest.fn(() => {}),
+  createTransferEngine: jest.fn().mockReturnValue({
+    transfer: jest.fn().mockReturnValue(Promise.resolve({})),
+  }),
+};
+
+jest.mock('@strapi/data-transfer', () => {
+  return mockDataTransfer;
+});
+
+const exportCommand = require('../transfer/export');
+
+const exit = jest.spyOn(process, 'exit').mockImplementation(() => {});
+jest.spyOn(console, 'error').mockImplementation(() => {});
+
+const defaultFileName = 'defaultFilename';
+
+jest.mock('../transfer/utils');
+
+describe('export', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    utils.getDefaultExportBackupName.mockReturnValue(defaultFileName);
+  });
+
+  it('uses path provided by user', async () => {
+    const filename = 'testfile';
+    await exportCommand({ output: filename });
+    expect(mockDataTransfer.createLocalFileDestinationProvider).toHaveBeenCalledWith(
+      expect.objectContaining({
+        file: { path: filename },
+      })
+    );
+    expect(utils.getDefaultExportBackupName).not.toHaveBeenCalled();
+    expect(exit).toHaveBeenCalled();
+  });
+
+  it('uses default path if not provided by user', async () => {
+    await exportCommand({});
+    expect(mockDataTransfer.createLocalFileDestinationProvider).toHaveBeenCalledWith(
+      expect.objectContaining({
+        file: { path: defaultFileName },
+      })
+    );
+
+    expect(utils.getDefaultExportBackupName).toHaveBeenCalled();
+    expect(exit).toHaveBeenCalled();
+  });
+
+  it('encrypts the output file if specified', async () => {
+    const encrypt = true;
+    await exportCommand({ encrypt });
+    expect(mockDataTransfer.createLocalFileDestinationProvider).toHaveBeenCalledWith(
+      expect.objectContaining({
+        encryption: { enabled: encrypt },
+      })
+    );
+    expect(exit).toHaveBeenCalled();
+  });
+
```

Comment:
```suggestion
  it('defaults encryption to true', async () => {
    await exportCommand({});
    expect(mockDataTransfer.createLocalFileDestinationProvider).toHaveBeenCalledWith(
      expect.objectContaining({
        encryption: { enabled: true },
      })
    );
    expect(exit).toHaveBeenCalled();
  });

  it('disables encryption when encrypt is false', async () => {
    const encrypt = false;
    await exportCommand({});
    expect(mockDataTransfer.createLocalFileDestinationProvider).toHaveBeenCalledWith(
      expect.objectContaining({
        encryption: { enabled: encrypt },
      })
    );
    expect(exit).toHaveBeenCalled();
  });
```

I think this would be a better way to test both. My fault, I never told you the defaults :)

Reply:
Yes, you're right. And it brings up the question of where the default filename should be determined. 
Right now it's set in the `export` _action_ whereas the encrypt and defaults defaults are set in the _command_, but maybe the filename should be set there too. And then we'll test that the defaults are set correctly later, [once we can more easily test the CLI itself](https://github.com/strapi/strapi/pull/14637).

Most of the other CLI features handle the defaults, and even additional user input, in the action, but I think it makes more sense (and I think it's also what you're saying?) to let Commander handle the inputs whenever possible, and the actions should be taking those inputs and acting on it.

Anyway, these tests are a good place to start, so go ahead and merge, and we can sync about this issue later :)